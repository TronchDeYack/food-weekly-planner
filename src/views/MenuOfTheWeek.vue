<template>
  <v-container>
    <div class="d-flex align-center justify-space-between mt-2">
      <h2>{{ $t('menu.title') }}</h2>
      <v-btn color="primary">{{ $t('menu.regenerate') }}</v-btn>
    </div>
    <v-row>
      <v-col>
        <MenuCard v-for="recipe in this.recipes" :key="recipe.id"
          :id="recipe.id" :title="recipe.title" :calories-level="recipe.caloriesLevel"
          :ingredients="recipe.ingredients" :minutes="recipe.minutes"
        />
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import MenuCard from '@/components/MenuCard.vue';
import { getRecipesOfTheWeek } from '@/api';

export default {
  name: 'menu-of-the-week',
  components: {
    MenuCard,
  },
  data() {
    return {
      recipes: [],
    };
  },
  mounted() {
    getRecipesOfTheWeek().then((recipes) => {
      this.recipes = recipes;
    });
  },
};
</script>
