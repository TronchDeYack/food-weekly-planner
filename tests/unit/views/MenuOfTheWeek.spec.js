import MenuOfTheWeek from '@/views/MenuOfTheWeek.vue';

import { createComponent } from '../../utils';

describe('MenuOfTheWeek', () => {
  it('Should render.', () => {
    const component = createComponent(MenuOfTheWeek);
    expect(component).toMatchSnapshot();
  });
});
