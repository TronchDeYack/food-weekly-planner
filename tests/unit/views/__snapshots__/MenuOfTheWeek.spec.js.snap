// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuOfTheWeek render Should render. 1`] = `
<v-container-stub tag="div">
  <div class="d-flex align-center justify-space-between mt-2">
    <h2>menu.title</h2>
    <v-btn-stub color="primary" tag="button" activeclass="" type="button">menu.regenerate</v-btn-stub>
  </div>
  <v-row-stub tag="div">
    <v-col-stub tag="div">
      <menucard-stub id="SOUPE-ID" title="Soupe Ã  la tomate" minutes="30" calorieslevel="low" ingredients="[object Object],[object Object]"></menucard-stub>
      <menucard-stub id="WELSH-ID" title="Welsh" minutes="45" calorieslevel="high" ingredients="[object Object],[object Object],[object Object],[object Object]"></menucard-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
